src/components/todoFooter.tsx

```diff
@@ -0,0 +1,109 @@
+import { useContext } from "react";
+import cn from "classnames";
+import { TodosContext } from "../services/Store";
+import { Status } from "../types/Status";
+import * as todoService from "../api/todos";
+import { ErrorText } from "../enums/errorText";
+
+export const TodoFooter: React.FC = () => {
+  const {
+    filter,
+    setFilter,
+    todos,
+    setTodos,
+    setLoadErrorMessage,
+    setIsSubmiting,
+    setClearButtonClicked,
+    setSelectedTodoId,
+  } = useContext(TodosContext);
+
+  const activeTodos = todos.filter((todo) => !todo.completed);
+
+  const handleClearComletedButton = () => {
+    setClearButtonClicked(true);
+    setIsSubmiting(true);
+
+    const completedTodoIds = [...todos]
```

Comment:
Consider using `Promise.allSettled()` here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled

Reply:
![image](https://github.com/mate-academy/react_todo-app-add-and-delete/assets/95346601/1e49503a-6f55-4c89-89c5-95c52d399c1a)
i tried to use it but i get the same error

Reply:
Feel free to ask for help in the chat. This function should handle all requests instead of one